lib/specinfra/ext/class.rb

```diff
@@ -2,7 +2,7 @@ class Class
   def subclasses
     result = []
     ObjectSpace.each_object(Class) do |k|
-      next if k.name.nil?
+      next unless k == k.ancestors.first #Skips if this is a singleton class
```

Comment:
It seems that singletons also have themselves as `ancestors.first`.

```ruby
require 'singleton'

class Foo
  include Singleton
end

p Foo.ancestos
```

This is the result.

```
$ ruby singleton.rb
[Foo, Singleton, Object, Kernel, BasicObject]
```

So how about this ?

```ruby
next if k.ancestors.include?(Singleton)
```